lib/restclient/request.rb

```diff
@@ -124,6 +124,7 @@ def initialize args
       else
         raise ArgumentError, "must pass :url"
       end
+      parse_url_with_auth!(url)
       @cookies = @headers.delete(:cookies) || args[:cookies] || {}
       @payload = Payload.generate(args[:payload])
       @user = args[:user]
```

Comment:
Looks like you're overriding the `@user` and `@password` you may have set on `parse_url_with_auth!`.